src/components/AssignModal/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import styles from './assign-modal.module.scss';
+import { X } from 'lucide-react';
+import Select from '@/components/elements/Select';
+
+interface AssignModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+interface FinalityOption {
+  label: string;
+  value: string;
+  description: string;
+}
+
+const finalityOptions: FinalityOption[] = [
+  {
+    label: 'Assign with Final finality',
+    value: 'final',
+    description:
+      'This means your region will be permanently assigned to a specific task.\n\nNo further modifications can be made to the region, which is why it will no longer appear on the regions page.\n\nThe benefit of choosing this option is that others can be confident that the Coretime cannot be unassigned from the task.\n\nAnother key benefit is that this option allows you to renew your core.',

```

Comment:
```suggestion
      'This means your region will be permanently assigned to a specific task.\n\nNo further modifications can be made to the region, which is why it will no longer appear on the regions page.\n\nBy choosing this option others can be confident that the Coretime cannot be unassigned from the task.\n\nAlso, with Final finality, the core becomes eligible for renewal.',

```